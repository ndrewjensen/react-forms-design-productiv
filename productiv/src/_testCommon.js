
const INITIAL_TODOS= [
  {
    id: 1,
    title: "Test1",
    description: "Test1 Description",
    priority: 1,
  },
  {
    id: 2,
    title: "Test2",
    description: "Test2 Description",
    priority: 2,
  },
     {
    id: 3,
    title: "Test3",
    description: "Test3 Description",
    priority: 3,
  }
];

export {INITIAL_TODOS}