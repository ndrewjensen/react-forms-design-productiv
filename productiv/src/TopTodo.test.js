import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";
import {INITIAL_TODOS} from "./_testCommon"


describe("productiv TopTodo", function () {
  it("renders without crashing", function () {
    render(<TopTodo todos={INITIAL_TODOS} />);
  });

  it("contains expected text", function () {
    const { container, debug } = render(<TopTodo todos={INITIAL_TODOS} />);

    expect(container.querySelector(".Todo")).toBeInTheDocument();
    expect(container).toContainHTML("<b>Test1</b>");
    expect(container).toContainHTML("<small> priority: 1</small>");
    expect(container).toContainHTML("<small>Test1 Description</small>");
  //   expect(container).toContainHTML(`
  //   <div><div class="Todo"><div><b>Test1</b><small> priority: 1</small></div><div><small>Test1 Description</small></div></div></div>`);
  });

  it("matches snapshot", function () {
    const { container } = render(<TopTodo todos={INITIAL_TODOS} />);
    expect(container).toMatchSnapshot();
  });
});
