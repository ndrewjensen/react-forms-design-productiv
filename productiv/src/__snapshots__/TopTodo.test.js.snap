// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv TopTodo matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Test1
      </b>
      <small>
         priority: 
        1
      </small>
    </div>
    <div>
      <small>
        Test1 Description
      </small>
    </div>
  </div>
</div>
`;
